RELEASE/scripts/autoscend/auto_quest_level_12.ash

```diff
@@ -1,5 +1,479 @@
+		//current code assumes it starts with 0 defeated out of 500 total. this is a guess.
+		if(auto_warSide() == "hippy")
+		{
+			enemiesRemaining = 500 - get_property("fratboysDefeated").to_int();
+		}
+		else
+		{
+			enemiesRemaining = 500 - get_property("hippiesDefeated").to_int();
+		}
+	}
+	else
+	{
+		if(auto_warSide() == "hippy")
+		{
+			enemiesRemaining = 1000 - get_property("fratboysDefeated").to_int();
+		}
+		else
+		{
+			enemiesRemaining = 1000 - get_property("hippiesDefeated").to_int();
+		}
+	}
+	return enemiesRemaining;
+}
+
+int auto_warKillsPerBattle()
+{
+	//returns how many enemies you will kill per battle at hippy-fratboy war at your current number of sidequests done.
+	return auto_warKillsPerBattle(auto_warSideQuestsDone());
+}
+
+int auto_warKillsPerBattle(int sidequests)
+{
+	//returns how many enemies you will kill per battle at hippy-fratboy war at a specified number of sidequests done.
+	
+	int kills = 2**sidequests;
+		
+	//Avatar of Sneaky Pete specific check
+	//TODO find correct spelling of mafia property for Rocket Launcher, I am currently guessing its spelling
+	if(get_property("peteMotorbikeCowling") == "Rocket Launcher")
+	{
+		kills += 3;
+	}
+	
+	//License to Adventure Path specific check
+	//TODO add it. it deals +3 kills per battle
+	
+	return kills;
+}
+
+int auto_warAdvReqCB()
+{
+	//how many adventures should it take to acquire a chaos butterfly
+	
+	if(get_property("chaosButterflyThrown").to_boolean() || item_amount($item[chaos butterfly]) > 0)
+	{
+		return 0;
+	}
+	if(canPull($item[chaos butterfly]))
+	{
+		return 0;
+	}
+	//4 enemies in [The Castle in the Clouds in the Sky (Ground Floor)] ~25% chance to encounter the one we want.
+	//roughly estimate 4 turns per possibility giant encounter. at base drop this means ~20 adv needed.
+	if(canYellowRay())
+	{
+		return 4;
+	}
+	
+	float expectedItemDropMulti = 1 + simValue("Item Drop")/100;
+	return ceil(20.0 / expectedItemDropMulti);
+}
+
+int auto_warAdvReqFarm()
+{
+	int advCost = 40;
+	
+	//TODO account for having done free fights in those zones
+	advCost -= $location[McMillicancuddy's Barn].turns_spent;
+	advCost -= $location[McMillicancuddy's Pond].turns_spent;
+	advCost -= $location[McMillicancuddy's Back 40].turns_spent;
+	advCost -= $location[McMillicancuddy's Other Back 40].turns_spent;
+	
+	//these paths cannot use butterfly
+	if(auto_my_path() == "Bees Hate You" || auto_my_path() == "Pocket Familiars")
+	{
+		return advCost;
+	}
+	
+	//chaos butterfly calculations
+	int advToGetCB = auto_warAdvReqCB();
+	if(get_property("chaosButterflyThrown").to_boolean() || item_amount($item[chaos butterfly]) > 0)
+	{
+		advCost -= 15;
+	}
+	else if(advToGetCB < 15)
+	{
+		advCost = advCost - 15 + advToGetCB;
+	}
+	
+	return advCost;
+}
+
+int auto_warPlan()
+{
+	//returns a bitwise of what is the best sidequest plan for the current run.
+	//due to bitwise, the first sidequest is sidequest number 0 and the last (sixth) sidequest is sidequest number 5, etc.
```

Comment:
Nitpick, these are bit fields. Bitwise is the term for the operators (like &, |, ^, << etc).

Reply:
fixed in PR #310